okhttp/src/main/kotlin/okhttp3/MediaType.kt

```diff
@@ -50,7 +52,12 @@ class MediaType private constructor(
   fun charset(defaultValue: Charset? = null): Charset? {
     val charset = parameter("charset") ?: return defaultValue
     return try {
-      Charset.forName(charset)
+      Charset.forName(charset).also {
+        if (it.name() != charset && !charsetRegex.matches(charset)) {
```

Comment:
wanna move the regex creation from where it is now to inside this call?

As is 100% of the time we class load MediaType we’re going to compile that pattern, and likely 0.000% of the time we’ll benefit from that work